projects/gnu.org/glibc/package.yml

```diff
@@ -0,0 +1,126 @@
+distributable:
+  url: https://ftp.gnu.org/gnu/glibc/glibc-{{ version.raw }}.tar.gz
+  strip-components: 1
+
+versions:
+  #TODO HTML listing: https://ftp.gnu.org/gnu/glibc/
+  # - 2.28
+  - 2.33
+
+runtime:
+  env:
+    PATH: ${{prefix}}/sbin:$PATH
+    CFLAGS: -Wl,--dynamic-linker={{prefix}}/lib/ld-{{ version.marketing }}.so
+
+build:
+  dependencies:
+    gnu.org/make: '>=3.79'
+    gnu.org/gawk: '>=3'
+    gnu.org/gcc: '*'
+    gnu.org/gettext: '*'
+    gnu.org/texinfo: '*'
+    gnu.org/bison: '*'
+    # gnu.org/patch: '*'
+    perl.org: '*'
+    # curl.se: '*'
+  working-directory: build
+  script: |
+    if test "{{hw.platform}}" != "linux"; then
+      echo "glibc is only supported on Linux"
+      mkdir -p "{{prefix}}"
+      touch {{prefix}}/linux-only
+      exit 0
+    fi
+
+    ../configure $ARGS
+    make all -j {{hw.concurrency}}
+    make install
+
+    cd {{prefix}}/bin
+    for s in $SCRIPTS; do
+      sed -i.bak 's|{{prefix}}|"$(cd "$(dirname "$0")/.." \&\& pwd)"|' $s
+      rm $s.bak
+    done
+  test:
+    make test
+  env:
+    SCRIPTS:
+      - catchsegv
+      - ldd
+      - mtrace
+      - sotruss
+      - tzselect
+      - xtrace
+    ARGS:
+      - --prefix={{ prefix }}
+      - --disable-debug
+      - --disable-dependency-tracking
+      - --disable-silent-rules
+      - --disable-werror
+      - --enable-obsolete-rpc
+      - --without-gd
+      - --without-selinux
+      - --enable-kernel=2.6.0
+      - --with-binutils={{deps.gnu.org/binutils.prefix}}/bin
+      # ChatGPT
+      - --disable-multi-arch
+    CFLAGS: -O2
+    linux/x86-64:
+      CFLAGS: -O2 -fPIC
+      CXXFLAGS: -O2 -fPIC
+      LDFLAGS: -pie
+
+test:
+  dependencies:
+    gnu.org/gcc: '*'
+  script: |
+    # Putting ourselves in the LD_LIBRARY_PATH breaks literally everything else...
+    export LD_LIBRARY_PATH=$(echo LD_LIBRARY_PATH | tr ':' '\n' | grep -v {{prefix}} | tr '\n' ':')
+    test "{{hw.platform}}" = "darwin" && exit 0
+    case "{{hw.arch}}" in
+      x86-64) ARCH=x86_64;;
+      aarch64) ARCH=aarch64;;
+      *)
+        echo "Unsupported architecture {{hw.arch}}"
+        exit 1
+        ;;
+    esac
+    gcc \
```

Comment:
`-nostdinc` & `-nostdlib`

Reply:
that looks less promising. you sure we want both?

Reply:
it'll stop it using the system headers and libraries… which is surely what we want?

Reply:
Possibly -nodefaultlibs?

https://stackoverflow.com/questions/10119520/g-static-initialization-and-nostdlib

Reply:
Oh ok I see. gcc considers glibc’s headers and libraries the system libraries…?

The goal is for gcc to only use glibc and tea libs, nothing from /usr/lib or /lib or /include or /usr/include etc.

Reply:
which means, I think, providing the init loader from libgcc. we'll see.